src/App.jsx

```diff
@@ -14,33 +19,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
+function sorteGoods (goods, sortedField, reversed) {
+  goods.sort((a, b) => {
+    switch (sortedField) {
+      case SORT_BY_ALPHABET:
+        return a.localeCompare(b);
+      case SORT_BY_LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  })
+
+  if (reversed) {
+    goods.reverse();
+  }
+
+  return goods;
+}
+export const App = () => {
+  const [sortedField,setSortedField] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  let goods = sorteGoods([...goodsFromServer], sortedField, reversed);
+
+  function reset () {
+    setSortedField('');
+    setReversed(false);
+    goods = [...goodsFromServer];
+  }
+  return (
   <div className="section content">
     <div className="buttons">
-      <button type="button" className="button is-info is-light">
+      <button
+        type="button"
+        className={
+          cn({
+            button: true,
+            'is-info': true,
+            'is-light': SORT_BY_ALPHABET !== sortedField
+          })
+        }
+        onClick={() => setSortedField(SORT_BY_ALPHABET)}
+      >
         Sort alphabetically
       </button>
 
-      <button type="button" className="button is-success is-light">
+      <button
+        type="button"
+        className={
+          cn({
+            button: true,
+            'is-success': true,
+            'is-light': SORT_BY_LENGTH !== sortedField
+          })
+        }
+        onClick={() => setSortedField(SORT_BY_LENGTH)}
+      >
         Sort by length
       </button>
 
-      <button type="button" className="button is-warning is-light">
+      <button
+        type="button"
+        className={
+          cn({
+            button: true,
+            'is-warning': true,
+            'is-light': !reversed
+          })
+        }
+        onClick={() => setReversed(!reversed)}
+      >
         Reverse
       </button>
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
+      {(sortedField || reversed) && (
+        <button
+          type="button"
+          className="button is-danger is-light"
+          onClick={reset}
+        >
+          Reset
+        </button>
+      )}
     </div>
 
     <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
+      {goods.map((good,i) =>(
+        <li key={good} data-cy="Good">{good}</li>
```

Comment:
Each element in the iteration should have a unique 'key' prop. Using the index 'i' as a key is not recommended if the list can change, but it is acceptable here since the list is static. However, using 'good' as the key is appropriate if 'good' is unique.